### PR TITLE
Add memory usage timelines

### PR DESCRIPTION
**Pull request by bduffany**
Example:

image

This screenshot is from running this python script as a test action. It repeatedly doubles the length of a string, starting from a string of length 1, sleeping for a bit before each doubling operation. It runs this whole procedure for 3 total iterations, sleeping a bit between each run.

```python
import gc
import time

for _ in range(3):
  s = "A"
  while len(s) < 100e6:
    s += s
    gc.collect()
    time.sleep(0.25)
  time.sleep(1)
  gc.collect()
```

### CODE REVIEW

File: enterprise/server/remote_execution/container/container.go [hunk 1]
```diff
@@ -335,9 +338,18 @@ func (s *UsageStats) TaskStats() *repb.UsageStats {
 }
 
 func (s *UsageStats) updateTimeline(now time.Time) {
-	if 8*(len(s.timeline.GetCpuSamples())+len(s.timeline.GetTimestamps())) > timeseriesSizeLimitBytes {
+	totalLength := 0
```
Comment (vanja-p) [inline enterprise/server/remote_execution/container/container.go:341]: nit: this loop is neat, but this is shorter and probably easier to read:
```
st := s.timeline
totalLength := st.GetTimestamps() + st.GetCpuSample() + st.GetMemoryKbSamples()
```

File: enterprise/server/remote_execution/container/container.go [hunk 2]
```diff
@@ -250,11 +250,14 @@ type UsageStats struct {
 // that TaskStats() can just return a view of the data rather than requiring a
 // full re-encoding.
 type timelineState struct {
```
Comment (vanja-p) [inline enterprise/server/remote_execution/container/container.go:252]: maybe document the units for the fields in this struct

**Review by vanja-p — APPROVED** [review]

File: enterprise/server/remote_execution/container/container.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/remote_execution/container/container.go:252]: Renamed the fields to include units

File: enterprise/server/remote_execution/container/container.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/remote_execution/container/container.go]: Done

**Pull request merged**